daniel_montecinos/space_challenge/src/main/java/com/tech_group/space_challenge/U2.java

```diff
@@ -0,0 +1,43 @@
+package com.tech_group.space_challenge;
+
+import java.util.Random;
+
+/**
+ * Represents a U1 Rocket object.
+ */
+public class U2 extends Rocket {
+    private static final Double COST;
+    private static final Double INITIAL_WEIGHT;
+    private static final Double MAX_WEIGHT;
+
+    static {
+        COST = 120000.0;
+        INITIAL_WEIGHT = 18000.0;
+        MAX_WEIGHT = 29000.0;
+    }
+
+    /**
+     * Creates and initializes a U1 Rocket.
+     */
+    public U2() {
+        super(COST, INITIAL_WEIGHT, MAX_WEIGHT);
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public boolean launch() {
+        changeToSurviveExplosion = new Random().nextDouble();
+        return changeToSurviveExplosion > 0.05 * (super.getWeight() / MAX_WEIGHT);
```

Comment:
Chance of launch explosion = **4%** * (cargo carried / cargo limit)
Chance of landing crash = **8%** * (cargo carried / cargo limit)

Please check the condition on the udacity page for U2

Reply:
done!!